include/odp/api/spec/event.h

```diff
@@ -209,6 +209,45 @@ void odp_event_free_multi(const odp_event_t event[], int num);
  */
 void odp_event_free_sp(const odp_event_t event[], int num);
 
+/**
+ * Event flow hash value
+ *
+ * Returns the hash value set in the event.
+ * Flow hash value set in the event is used by scheduler to distribute the
+ * event across different flows.
+ *
+ * @param	event	Event handle
+ *
+ * @return		Hash value
+ *
+ * @note The hash algorithm and the header fields defining the flow (therefore
+ * used for hashing) is platform dependent. The initial event flow hash
+ * generated by the HW will be same for flow hash generated for packets.
+
+ * @note The returned hash is either the platform generated (if any), or if
+ * odp_event_flow_hash_set() were called then the value set there.
+ */
+uint32_t odp_event_flow_hash(odp_event_t event);
+
+/**
+ * Set event flow hash value
+ *
+ * Store the event flow hash for the event and sets the flow hash flag.
+ * When scheduler is configured as flow aware, schedule queue synchronization
+ * will be based on flow within each queue.
+ * When scheduler is configured as flow unaware, event flow hash is ignored by
+ * the implementation.
+ * The value of flow hash should not be greater than the max flow count
```

Comment:
This comment also supports calling this a flow id since hashes don't have an associated ordering. An id, by contrast, can be constrained to fall within a given range of valid values.

Reply:
ack